### DISCUSSION TITLE
Custom Hook sent Email

### DISCUSSION THREAD

**Discussion by kaiquefeng** [directus/directus#12290] [Archive]
Is it possible to use a data object sent in the custom hook? to display different data in the email body

**Comment by rijkvanzanten** [comment]
If you're sending a custom email from a hook, you can make the message body say whatever you want 👍🏻

**Reply by kaiquefeng** [reply]
for the email template to receive the body data, how can I pass this information from the hook to the email body?

**Reply by rijkvanzanten** [reply]
Depends on what you're doing in your hook! You can pull any data from the system as you have access to the database in a hook, so the sky is the limit

**Reply by kaiquefeng** [reply]
```js
import axios from 'axios';

export default ({ filter }, { services, exceptions }) => {
	const { MailService } = services;
	const { ServiceUnavailableException, ForbiddenException } = exceptions;

	// Sync with external recipes service, cancel creation on failure
	filter('items.create', async (input, { collection }, { schema }) => {
		if (collection !== 'recipes') return input;

		const mailService = new MailService({ schema });

		try {
			await axios.post('https://example.com/recipes', input);
			await mailService.send({
				to: 'person@example.com',
				template: {
					name: 'item-created',
					data: {
						collection: collection,
					},
				},
			});
		} catch (error) {
			throw new ServiceUnavailableException(error);
		}

		input.syncedWithExample = true;

		return input;
	});
};

```

I using model hook to sent


Inside the template ( email.liquid ) I can use a variable like: ( {{ data }} ) to receive a body with the content that comes from the hook

**Reply by rijkvanzanten** [reply]
Ah yeah, so you can just pass any other information on to that `data` field then to make it available to your email template 🙂 

```
			await mailService.send({
				to: 'person@example.com',
				template: {
					name: 'item-created',
					data: {
						collection: collection,
						whatever: 'you want'
					},
				},
			});
```

**Reply by kaiquefeng** [reply]
Passing it this way, inside the email template I can access the variable like: ( {{ whatever }} or something?

**Reply by rijkvanzanten** [reply]
Yup!

**Reply by kaiquefeng** [reply]
thanks!!!
